src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -2262,21 +2262,243 @@ void M(C c)
             ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
             : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
 
-        if (inSource)
+        comp.VerifyDiagnostics(
+            // (3,12): error CS0619: 'C' is obsolete: 'error'
+            //     void M(C c)
+            Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "C").WithArguments("C", "error").WithLocation(3, 12)
+            );
+    }
+
+    [Theory, CombinatorialData]
+    public void WithObsolete_ReverseOrder(bool inSource)
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+[System.Obsolete("error", true)]
+public class C
+{
+}
+""";
+
+        var src = """
+class D
+{
+    void M(C c)
+    {
+    }
+}
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics(
+            // (3,12): error CS0619: 'C' is obsolete: 'error'
+            //     void M(C c)
+            Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "C").WithArguments("C", "error").WithLocation(3, 12)
+            );
+    }
+
+    public enum ObsoleteKind
+    {
+        Deprecated,
+        Obsolete,
+        WindowsExperimental,
+        Experimental
+    }
+
+    [Theory, CombinatorialData]
+    public void PriorityOrder(ObsoleteKind one, ObsoleteKind other, bool inSource, bool reversed)
```

Comment:
>reversed [](http://example.com/codeflow?start=88&length=8)

Do we really need this parameter? Doesn't combinatorial set include all possible orders?

Reply:
Thanks. It's indeed not needed